Mathlib/Data/Set/Basic.lean

```diff
@@ -130,6 +130,11 @@ instance PiSetCoe.canLift' (ι : Type u) (α : Type v) [Nonempty α] (s : Set ι
     CanLift (s → α) (ι → α) (fun f i => f i) fun _ => True :=
   PiSetCoe.canLift ι (fun _ => α) s
 
+@[simp]
+lemma Disjoint.inter_eq_empty {s t : Set α} (h : Disjoint s t) : s ∩ t = ∅ := by
```

Comment:
We already have `Disjoint.inter_eq`. If you tag this lemma as `@[simp]`, then `simp` will try to prove `Disjoint s t` every time it meets `s ∩ t` in a formula. Do we really want this?

Reply:
I see that my question here goes contrary to what @eric-wieser and @YaelDillies  suggested earlier. Let's bench to see if it slows down compilation.